### PR TITLE
Refactor to Typescript + Project screens + Everything else

### PR DESCRIPTION
**Pull request by maartenvn**
This is a full rewrite of the core of the project to Typescript.

- Typescript is now used inside the components itself using `vue-class-component` and `vue-property-decorators` (this is what Vue CLI 3 recommends when using the class syntax)
- [EchoFetch](https://www.npmjs.com/package/echofetch) is used for requests to the API. This is a wrapper around Axios that supports easy requests using services and has great Vue bindings.
- A better and more modern project page. The data itself is still fake date until my requested backend changes are done.
- Adds a better API for opening modals & snackbars
- Adds a error handling API that follows the requested format:
```javascript
// A general error returns errors that are not linked to any input field.
generalErrors: [
    {
        message: "Something general has occurred."
    }
],

// A field error returns an error that is linked to a body input field. 
// (eg name or description when creating a project)
inputErrors: [
     {
        input: "name",
        message: "Name cannot be longer than 3 characters."
     }
]
```

### CODE REVIEW

**Comment by maartenvn** [comment]
I have added a base for i18n. I would support both English & Dutch since G2 will be the key access point for all Zeus projects, so supporting multiple languages would be great.

File: src/i18n/nl.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+                "description": "Beschrijving"
+            }
+        }
+    },
+
+    "login": {
+        "title": "Inloggen",
+        "desc": "Kies een van de volgende opties om verder te gaan.",
+        "zeus": "Inloggen met Zeus WPI",
+        "github": "Inloggen Github"
```
Comment (hanneskt) [inline src/i18n/nl.json]:
```suggestion
        "github": "Inloggen met Github"
```

**Comment by maartenvn** [comment]
The frontend does no longer work with the latest version of the backend (as of right now).
Use a mock server with the OpenAPI specification (g2-backend/openapi) to run the frontend.

**Review by mcbloch — APPROVED** [review]
Merging this to master to keep it more up to date with the actual development.

**Pull request merged**